### PR TITLE
3171 describe get api v2 viz id viz

### LINKED ISSUE #3171: Move test suite  describe 'GET /api/v2/viz/:id/viz'

### PR DESCRIPTION
**Pull request by juanignaciosl**
@rafatower , @Kartones CR this. Closes #3171, please.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3020/))

**Comment by juanignaciosl** [comment]
@javisantana CR you too, please, since I've also edited the function you used for url switching.

File: app/models/carto/visualization.rb [hunk 1]
```diff
@@ -67,18 +74,58 @@ def likes_by_user_id(user_id)
...
+  end
+
   def get_related_tables
     return [] unless map
     map.carto_and_torque_layers.flat_map { |layer| layer.affected_tables.map { |t| t.service } }.uniq
   end
 
   def is_viewable_with_link?
-    is_public? || self.privacy == PRIVACY_LINK
+    self.privacy == PRIVACY_LINK
```
Comment (rafatower) [inline app/models/carto/visualization.rb]: Watch out, even if this is not nicely written I think it should return true if privacy == public || privacy == link
I'd use a case-when statement for this kind of table-like checks

File: app/models/carto/visualization.rb [hunk 2]
```diff
@@ -67,18 +74,58 @@ def likes_by_user_id(user_id)
...
+  end
+
   def get_related_tables
     return [] unless map
     map.carto_and_torque_layers.flat_map { |layer| layer.affected_tables.map { |t| t.service } }.uniq
   end
 
-  def is_viewable_with_link?
-    is_public? || self.privacy == PRIVACY_LINK
+  def is_link_privacy?
```
Comment (juanignaciosl) [inline app/models/carto/visualization.rb]: Fair enough. I've changed method name to match what it did. It's a private method used by public ones which the meaning you spoke about.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3024/))

**Comment by Kartones** [comment]
looks ok :+1:

**Comment by javisantana** [comment]
:es:

**Comment by rafatower** [comment]
Looks good :+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3028/))

**Pull request merged**